text/0000-build-observability.md

```diff
@@ -0,0 +1,358 @@
+      "scopeSpans": [
+        {
+          "scope": {},
+          "spans": [
+            {
+              "traceId": "",
+              "spanId": "",
+              "parentSpanId": "",
+              "name": "buildpack-detect",
+              "startTimeUnixNano": "1581452772000000321",
+              "endTimeUnixNano": "1581452773000000789",
+              "droppedAttributesCount": 2,
+              "events": [
+                {
+                  "timeUnixNano": "1581452773000000123",
+                  "name": "detect-pass"
+                }
+              ],
+              "attributes": [
+                {
+                  "key": "buildpack-id",
+                  "value": {
+                    "stringValue": "heroku/nodejs-engine"
+                  }
+                }
+              ],
+              "droppedEventsCount": 1
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+
+### Buildpack telemetry files
+
+During a buildpack's `detect` or `build` execution, a buildpack may emit
+an OpenTelemetry File Export with tracing data to `/cnb/telemetry/#{buildpack-id}.jsonl`
+with contents like this:
+
+```json
+{
+  "resourceSpans": [
+    {
+      "resource": {
+        "attributes": [
+          {
+            "key": "lifecycle.version",
+            "value": {
+              "stringValue": "0.17.1"
+            }
+          }
+        ]
+      },
+      "scopeSpans": [
+        {
+          "scope": {},
+          "spans": [
+            {
+              "traceId": "",
+              "spanId": "",
+              "parentSpanId": "",
+              "name": "buildpack-detect",
+              "startTimeUnixNano": "1581452772000000321",
+              "endTimeUnixNano": "1581452773000000789",
+              "droppedAttributesCount": 2,
+              "events": [
+                {
+                  "timeUnixNano": "1581452773000000123",
+                  "name": "detect-pass"
+                }
+              ],
+              "attributes": [
+                {
+                  "key": "buildpack-id",
+                  "value": {
+                    "stringValue": "heroku/nodejs-engine"
+                  }
+                }
+              ],
+              "droppedEventsCount": 1
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Lifetime
+
+Telemetry files may be written at any point during the build, so that they
+are persisted in cases of failures to detect, failures to build, process
+terminations, or crashes. The `jsonl` format allows telemetry libraries to
+safely append additional json objects to the end of a telemetry file, so
+telemetry data can be flushed to the file frequently. Telemetry files should
+not be truncated or deleted so that telemetry processing by a platform can
+happen during or after a build. Telemetry files should not be included in the
+build result, as they are not relevant, and would likely negatively impact
+image size and reproduceability.
+
+### Access
```

Comment:
I'd like to see a couple statements in this section about _intent_. Mainly:

* Buildpacks should not use the telemetry files to pass materiel to another buildpack, or to implement any kind of contract between buildpacks.
*  A platform may access these files at any time during the build in order to evacuate them for safe keeping in the event of a build failure. 

or something like that?

Reply:
Added an intent section here: https://github.com/buildpacks/rfcs/blob/6438aed9b16f226a6f54454d6a12dd0871f266d1/text/0000-build-observability.md?plain=1#L258. 

I covered the point about access in the consumption section: https://github.com/buildpacks/rfcs/blob/6438aed9b16f226a6f54454d6a12dd0871f266d1/text/0000-build-observability.md?plain=1#L234.